PWGHF/D2H/Macros/efficiency.py

```diff
@@ -0,0 +1,389 @@
+                linestyle=LINESTYLE[key])
+
+#pylint: disable=too-many-locals, too-many-branches, too-many-statements
+def main(
+    cfg: dict,
+    batch: bool) -> None:
+    """
+    Main function
+
+    Parameters
+    -----------------
+    - config: dictionary with config read from a yaml file
+    - batch: bool to suppress video output
+    """
+
+    set_global_style(padleftmargin=0.14, padbottommargin=0.12, titlesize=0.045, labelsize=0.04)
+
+    # input configuration
+    name_file = cfg['input']['filename']
+    name_tree = cfg['input']['treename']
+    file = TFile.Open(name_file)
+    name_tree = enforce_trailing_slash(name_tree)
+
+    # fill dictionaries with histograms
+    dic_rec, dic_gen = {}, {}
+    for key, histoname in cfg['input']['histoname']['reconstructed'].items():
+        if histoname is None:
+            dic_rec[key] = None
+        else:
+            dic_rec[key] = file.Get(name_tree + histoname)
+    for key, histoname in cfg['input']['histoname']['generated'].items():
+        if histoname is None:
+            dic_gen[key] = None
+        else:
+            dic_gen[key] = file.Get(name_tree + histoname)
+
+    #  configure possible cut on rapidity
+    rapidity_cut = cfg['rapidity']['cut']
+    axis_rapidity = cfg['rapidity']['axis']
+
+    # output configuration
+    out_dir = enforce_trailing_slash(cfg['output']['dir'])
+    out_label = cfg['output']['plots']['label']
+    name_axis = cfg['output']['plots']['y_axis']['name']
+    min_y_axis = cfg['output']['plots']['y_axis']['min']
+    max_y_axis = cfg['output']['plots']['y_axis']['max']
+    log_y_axis = cfg['output']['plots']['y_axis']['log_scale']
+    if name_axis is None:
+        name_axis = '#varepsilon' # default value
+    pt_min, pt_max = 0., 35. # default values
+    title = ';#it{p}_{T} (GeV/#it{c});' + name_axis + ';'
+    overlap = enforce_list(cfg['output']['plots']['overlap'])
+
+    # output save options
+    save_tefficiency = cfg['output']['save']['TEfficiency']
+    save_th1 = cfg['output']['save']['TH1']
+    save_tcanvas_individual = cfg['output']['save']['TCanvas']['individual']
+    save_tcanvas_overlap = cfg['output']['save']['TCanvas']['overlap']
+    extension = enforce_list(cfg['output']['save']['TCanvas']['extension'])
+
+    if os.path.isdir(out_dir):
+        print((f'\033[93mWARNING: Output directory \'{out_dir}\' already exists,'
+                    ' overwrites possibly ongoing!\033[0m'))
+    else:
+        os.makedirs(out_dir)
+
+    # plots watermark configuration
+    latex = TLatex()
+    latex.SetNDC()
+    latex.SetTextSize(0.035)
+    latex.SetTextAlign(13)  # align at top
+    latex.SetTextFont(42)
+    watermark = cfg['output']['plots']['watermark']
+
+    # open output file
+    out_file = TFile(out_dir + out_label + '.root', 'recreate')
+
+    # compute efficiencies
+    efficiencies = {}
+    is_retrieved_pt_interval = False
+    for (key, h_rec), (_, h_gen) in zip(dic_rec.items(), dic_gen.items()):
```

Comment:
What about adding a couple of protections/warning here, checking wheter the cases
`h_rec is None`
and
`h_gen is None`
are satisfied?